### PR TITLE
Squeezelite package and service init

### PR DESCRIPTION
**Pull request by pwetzel**
###### Motivation for this change

This PR adds a package and service definition for Squeezelite, a headless Squeezebox emulator. This software works with Logitech Media Server for synchronized multi-device playing of local and "cloud" music sources.
###### Things done
- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/modules/services/audio/squeezelite.nix [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+in {
+
+  ###### interface
+
+  options = {
+
+    services.squeezelite= {
+
+      enable = mkOption {
```
Comment (joachifm) [inline nixos/modules/services/audio/squeezelite.nix]: `mkEnableOption` is a short-hand for this. Note that it expands to something like "Whether to enable foo", so take that into account when choosing its argument.

File: pkgs/applications/audio/squeezelite/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{ stdenv, fetchgit, alsaLib, flac, libmad, libvorbis, faad2, mpg123 }:
+
+stdenv.mkDerivation {
+  name = "squeezelite-git-2016-05-27";
+
+  src = fetchgit {
```
Comment (joachifm) [inline pkgs/applications/audio/squeezelite/default.nix]: Does this use git submodules? If not, you can use `fetchFromGitHub`, which expands an arbitrary revision to a url to a autogenerated tarball.

File: pkgs/applications/audio/squeezelite/default.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    sha256 = "15ihx2dbp4kr6k6r50g9q5npqad5zyv8nqf5cr37bhg964syvbdm";
+  };
+
+  buildInputs = [ stdenv alsaLib flac libmad libvorbis faad2 mpg123 ];
+
+  buildPhase = ''
+    make
+  '';
+
+  doCheck = false;
```
Comment (joachifm) [inline pkgs/applications/audio/squeezelite/default.nix]: This is the default

File: pkgs/applications/audio/squeezelite/default.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+stdenv.mkDerivation {
+  name = "squeezelite-git-2016-05-27";
+
+  src = fetchgit {
+    url = "https://github.com/ralph-irving/squeezelite.git";
+    rev = "e37ed17fed9e11a7346cbe9f1e1deeccc051f42e";
+    sha256 = "15ihx2dbp4kr6k6r50g9q5npqad5zyv8nqf5cr37bhg964syvbdm";
+  };
+
+  buildInputs = [ stdenv alsaLib flac libmad libvorbis faad2 mpg123 ];
```
Comment (joachifm) [inline pkgs/applications/audio/squeezelite/default.nix]: Is it really necessary to add `stdenv` here?

File: pkgs/applications/audio/squeezelite/default.nix [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+  buildPhase = ''
```
Comment (joachifm) [inline pkgs/applications/audio/squeezelite/default.nix]: Is this strictly necessary?

File: nixos/modules/misc/ids.nix [hunk 6]
```diff
@@ -269,6 +269,7 @@
       nzbget = 245;
       mosquitto = 246;
       toxvpn = 247;
+      squeezelite = 248;
```
Comment (groxxda) [inline nixos/modules/misc/ids.nix]: Please also add a line to the list of gids to mark gid 248 as reserved

File: pkgs/applications/audio/squeezelite/default.nix (hunk 5 above)
Comment (pwetzel) [inline pkgs/applications/audio/squeezelite/default.nix]: No. It caused a few less warnings in the build output, but it does work fine with the default builder. I removed it.

File: nixos/modules/services/audio/squeezelite.nix (hunk 1 above)
Comment (pwetzel) [inline nixos/modules/services/audio/squeezelite.nix]: Thanks, switched to that.

File: pkgs/applications/audio/squeezelite/default.nix (hunk 2 above)
Comment (pwetzel) [inline pkgs/applications/audio/squeezelite/default.nix]: Wasn't aware of that function, thanks.

File: pkgs/applications/audio/squeezelite/default.nix (hunk 3 above)
Comment (pwetzel) [inline pkgs/applications/audio/squeezelite/default.nix]: Thanks.

File: pkgs/applications/audio/squeezelite/default.nix (hunk 4 above)
Comment (pwetzel) [inline pkgs/applications/audio/squeezelite/default.nix]: I started from a nix-shell definition and forgot to remove it when making the package. Thanks.

File: nixos/modules/misc/ids.nix (hunk 6 above)
Comment (pwetzel) [inline nixos/modules/misc/ids.nix]: Done, thanks.

**Comment by pwetzel** [comment]
Thanks very much for all the feedback! I believe I've incorporated everything that was raised.

**Comment by pwetzel** [comment]
@joachifm, is there anything else I should do to get this merged?

**Comment by joachifm** [comment]
Looks just about ready to me. Thank you.

**Pull request merged**